tectonic/synctex.c

```diff
@@ -342,105 +342,18 @@ void synctex_start_input(void)
-        while (tmp > the_real_syncname) {
-            --tmp;
-            if (*tmp == '.') {
-                *tmp = (char)0;    /* end the string here */
-                break;
-            }
-        }
-        strcat(the_real_syncname, synctex_suffix);
-        /* allways remove the synctex output file before renaming it, windows requires it. */
-        if (0 != remove(the_real_syncname) && errno == EACCES) {
-            fprintf(stderr,
-                    "SyncTeX: Can't remove %s (file is open or read only)\n",
-                    the_real_syncname);
-        }
-        if (synctex_ctxt.file) {
-            if (synctex_ctxt.flags.not_void) {
-                synctex_record_postamble();
-                /* close the synctex file */
-                xfclose(synctex_ctxt.file, synctex_ctxt.busy_name);
-                synctex_ctxt.file = NULL;
-                /*  renaming the working synctex file */
-                if (0 == rename(synctex_ctxt.busy_name, the_real_syncname)) {
-                    if (log_opened) {
-                        tmp = the_real_syncname;
-                        printf("\nSyncTeX written on %s.", tmp);
-                        tmp = NULL;
-                    }
-                } else {
-                    fprintf(stderr, "SyncTeX: Can't rename %s to %s\n",
-                            synctex_ctxt.busy_name, the_real_syncname);
-                    remove(synctex_ctxt.busy_name);
-                }
-            } else {
-                /* close and remove the synctex file because there are no pages of output */
-                xfclose(synctex_ctxt.file, synctex_ctxt.busy_name);
-                synctex_ctxt.file = NULL;
-                remove(synctex_ctxt.busy_name);
-            }
-        }
-    } else if (tmp = gettexstring(job_name)) {
-        size_t len = strlen(tmp);
-        /*  There was a problem with the output.
-         We just try to remove existing synctex output files
-         including the busy one. */
-        the_real_syncname = xmalloc((size_t)
-                                    (len + strlen(synctex_suffix)
-                                     + strlen(synctex_suffix_gz) + 1));
-        if (!the_real_syncname) {
-            free(tmp);
-            synctexabort();
-            return;
-        }
-        strcpy(the_real_syncname, tmp);
-        free(tmp);
-        tmp = NULL;
-        strcat(the_real_syncname, synctex_suffix);
-        remove(the_real_syncname);
-        strcat(the_real_syncname, synctex_suffix_gz);
-        remove(the_real_syncname);
-        if (synctex_ctxt.file) {
-            /* close the synctex file */
-            xfclose(synctex_ctxt.file, synctex_ctxt.busy_name);
+    if (synctex_ctxt.file) {
+        /* We keep the file even if no tex output is produced. I assume that this means that
+         * there was an error and tectonic will not save anything anyway. */
+        /* if (synctex_ctxt.flags.not_void) { */
```

Comment:
Please remove the commented `if` statement and revise the comment above to say something like "... even if no tex output is produced (synctex_ctxt.flags.not_void = 0). I assume ..."